pettingzoo/utils/conversions.py

```diff
@@ -73,29 +73,35 @@ def parallel_to_aec(par_env: ParallelEnv) -> AECEnv:
         return ordered_env
 
 
-def turn_based_aec_to_parallel(aec_env: AECEnv) -> ParallelEnv:
+def turn_based_aec_to_parallel(
+    aec_env: AECEnv[AgentID, ObsType, Optional[ActionType]]
+) -> ParallelEnv[AgentID, ObsType, Optional[ActionType]]:
     if isinstance(aec_env, parallel_to_aec_wrapper):
         return aec_env.env
     else:
         par_env = turn_based_aec_to_parallel_wrapper(aec_env)
         return par_env
 
 
-def to_parallel(aec_env: AECEnv) -> ParallelEnv:
+def to_parallel(
+    aec_env: AECEnv[AgentID, ObsType, ActionType]
+) -> ParallelEnv[AgentID, ObsType, ActionType]:
     warnings.warn(
         "The `to_parallel` function is deprecated. Use the `aec_to_parallel` function instead."
     )
     return aec_to_parallel(aec_env)
 
 
-def from_parallel(par_env: ParallelEnv) -> AECEnv:
+def from_parallel(
+    par_env: ParallelEnv[AgentID, ObsType, Optional[ActionType]]
+) -> AECEnv[AgentID, ObsType, Optional[ActionType]]:
     warnings.warn(
         "The `from_parallel` function is deprecated. Use the `parallel_to_aec` function instead."
     )
     return parallel_to_aec(par_env)
 
 
-class aec_to_parallel_wrapper(ParallelEnv):
+class aec_to_parallel_wrapper(ParallelEnv[AgentID, ObsType, ActionType]):
```

Comment:
This occurrence of it has ActionType rather than Optional[ActionType], looks like you just forgot to add it but maybe it's intentional for some reason?

Reply:
It's only necessary from ParallelEnv -> AECEnv, but not the other way around.